email-mirror: Move postfix email mirror integration to separate script.

kkanahin opened the pull request:
- Add script wrapper to comunicate postfix pipe with tornado we server
  over HTTP protocol. It uses shared_secret authentication mode.
- Add tornado view to process messages from email mirror server.
- Clean managment command `email-mirror`. Leае just functional
  for cron email processing.
- Add routes for new tornado view.
- Change pipe script in master process postfix config template.
- Add tests.

Fixes #2421

kkanahin commented:
@timabbott Could you review this PR, please.

timabbott commented:
This looks great, except that we want Django, not Tornado, to be handling the requests (Tornado should only be used for some special-purpose use cases).  So we need to refactor `internal_notify_view` to, say, take an argument specifying whether it's for Tornado or not (you can see how to do arguments to decorators via `api_key_only_webhook_view`).  

I went ahead and did that refactoring in cbc731963f9508c86d6554b9c23e24a6b2eb0b13; you should be able tor rebase this onto those changes to make this work using Django as the server backend using  `internal_notify_view(False)` as the decorator.

kkanahin commented:
@timabbott Thanks a lot for review. Will change destination server to django.

kkanahin commented:
@timabbott I've updated this PR. I've moved view to django and added advanced options to script for more flexible configuration.  Could you please check it again.

timabbott commented on puppet/zulip/files/postfix/master.cf:
```diff
@@ -111,4 +111,4 @@ mailman   unix  -       n       n       -       -       pipe
   ${nexthop} ${user}
 zulip unix  -       n       n       -       -       pipe
-  flags= user=zulip argv=/home/zulip/deployments/current/manage.py email_mirror ${original_recipient}
+  flags= user=zulip argv=/home/zulip/deployments/current/scripts/email_mirror_postfix -r ${original_recipient}
```
This doesn't match the filename of the script itself.

kkanahin replied:
Oh yes, I forgot to change it after renaming.

kkanahin commented:
@timabbott I've renamed filename in postfix config, thank you.

timabbott commented:
I made a few changes:
* Moved this to scripts/lib to keep the root scripts/ uncluttered
* Set `verify=False` automatically if running against https://127.0.0.1, since then SSL is neither required nor effective.

and merged this as e3e52e728465eddafecd7c8a91f6ed2825f70483, thanks @kkanahin!!

One follow-up issue is that the startup perf of the new script is slower than it should be in prod (~300ms), since it has some expensive imports (requests is ~100ms, and setting up the Zulip venv is ~200ms).  I think we should at least get rid of the venv dependency if we can...

kkanahin commented:
Thanks, @timabbott.  I will think about performance.
